File: Application/DTOs/ReportItemDto.cs
namespace Application.DTOs;

public class ReportItemDto
{
    public string MedicationName { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public string Unit { get; set; } = string.Empty;
    public string Reason { get; set; } = string.Empty;
}

File: frontend/lib/data/dtos/report_item_dto.dart
class ReportItemDto {
  final String medicationName;
  final int quantity;
  final String unit;
  final String reason;

  ReportItemDto({
    required this.medicationName,
    required this.quantity,
    required this.unit,
    required this.reason,
  });

  factory ReportItemDto.fromJson(Map<String, dynamic> json) {
    return ReportItemDto(
      medicationName: json['medicationName'] ?? '',
      quantity: json['quantity'] ?? 0,
      unit: json['unit'] ?? '',
      reason: json['reason'] ?? '',
    );
  }
}